### DISCUSSION TITLE
Is it normal the MQTT to disconnect immediately after publish?

### DISCUSSION THREAD

**Discussion by theater** [emsesp/EMS-ESP32#1104] [Q&A] [answered]
Hi,
I just updated from the old hardware to the new EMS-ESP32 with firmware 3.5.0 stable.
I find it weird that the console logs are filled with MQTT connected / MQTT disconnected.
The logs look like this if I enable debug:
000+00:20:39.681 I 906: [mqtt] MQTT connected
000+00:20:39.681 D 907: [mqtt] Subscribing to topic 'homeassistant/+/home_bosch_CW7000/#'
000+00:20:39.782 D 908: [mqtt] Publishing topic home/bosch_CW7000/heartbeat (#3549, retain=0, retry=1, size=310, pid=1)
000+00:20:39.883 D 909: [mqtt] Subscribing to topic 'home/bosch_CW7000/system/#'
000+00:20:39.984 D 910: [mqtt] Subscribing to topic 'home/bosch_CW7000/dallasssensor/#'
000+00:20:40.085 D 911: [mqtt] Subscribing to topic 'home/bosch_CW7000/analogsensor/#'
000+00:20:40.186 D 912: [mqtt] Subscribing to topic 'home/bosch_CW7000/boiler/#'
000+00:20:40.287 D 913: [mqtt] Subscribing to topic 'home/bosch_CW7000/thermostat/#'
000+00:20:40.388 D 914: [mqtt] Publishing topic home/bosch_CW7000/status (#3555, retain=1, retry=1, size=6, pid=1)
000+00:20:40.489 D 915: [mqtt] Publishing topic home/bosch_CW7000/boiler_data_ww (#3556, retain=0, retry=1, size=461, pid=1)
000+00:20:40.590 D 916: [mqtt] Publishing topic home/bosch_CW7000/boiler_data (#3557, retain=0, retry=1, size=711, pid=1)
000+00:20:40.691 D 917: [mqtt] Publishing topic home/bosch_CW7000/thermostat_data (#3558, retain=0, retry=1, size=936, pid=1)
000+00:20:40.798 W 918: [mqtt] MQTT disconnected: TCP

Is this a normal behavior of the firmware or something is wrong with the configuration. The Wi-fi seems stable as I don't loose my connection to the telnet console.
The data is published successfully and the queue always around 0.

Best regards,
Konstantin

**Comment by MichaelDvP** [comment]
No, this is not normal behaviour. Check your mqtt-broker.

**Comment by proddy** [comment, marked as answer]
could be a conflicting MQTT client-id? Without seeing your settings it's hard to find out what is wrong. See https://emsesp.github.io/docs/Support/

As Michael said, turn on verbose logging on your MQTT broker to see what is happening server side

**Comment by theater** [comment]
Thank you so much. That was it. I set the same id for the new device and
left the old still powered on.
I checked yesterday the logs of both the mosquitto in verbose mode and the
ems-esp but couldn't figure it out.
Now everything is working fine.

Thank you again. Will send this info to bbqees, so he may want to add this
to the FAQ.
